compiler-core/src/erlang/tests/numbers.rs

```diff
@@ -32,3 +32,28 @@ pub fn main() {
 "#
     );
 }
+
+#[test]
+fn numbers_with_scientific_notation() {
+    assert_erl!(
+        r#"
+const i = 100.001e523
+const j = -100.001e-523
+
+const k = 100.001e1_230
+const l = -100.001e-1_230
+
+const m = 100.001e123_456_789
+const n = -100.001e-123_456_789
+
+pub fn main() {
```

Comment:
It's to test the correct Erlang is generated

Reply:
So the variable references in main() are required because else the constants don't exist in erl representation? If that's the case sorry for the noise.

Reply:
Yup that's right 👍